tw-tasks-core/src/main/java/com/transferwise/tasks/helpers/kafka/partitionkey/IPartitionKeyStrategy.java

```diff
@@ -0,0 +1,12 @@
+package com.transferwise.tasks.helpers.kafka.partitionkey;
+
+import com.transferwise.tasks.domain.BaseTask;
+
+/**
+ * Defines a strategy to provide a partition key. The partition key will be used by tw-tasks to send messages to Kafka while triggering a job.
+ */
+public interface IPartitionKeyStrategy {
+
+  String createPartitionKey(BaseTask task);
```

Comment:
I wonder if there are use cases, where someone wants to set partition number explicitly.

If there are, the interface could return some kind of tuple of "partitionKey,partition", where "partition" is preferred when it is set.

In any case, this is simple to add later as well.